src/sql-pretty/src/doc.rs

```diff
@@ -94,6 +94,470 @@ impl Pretty {
+                ")",
+            ));
+        }
+
+        RcDoc::intersperse(docs, Doc::line()).group()
+    }
+
+    pub(crate) fn doc_create_cluster<'a, T: AstInfo>(
+        &'a self,
+        v: &'a CreateClusterStatement<T>,
+    ) -> RcDoc<'a> {
+        let mut docs = Vec::new();
+
+        // CREATE CLUSTER name
+        docs.push(nest_title("CREATE CLUSTER", self.doc_display_pass(&v.name)));
+
+        // OPTIONS (...)
+        if !v.options.is_empty() {
+            docs.push(bracket(
+                "(",
+                comma_separate(|o| self.doc_display_pass(o), &v.options),
+                ")",
+            ));
+        }
+
+        // FEATURES (...)
+        if !v.features.is_empty() {
+            docs.push(bracket(
+                "FEATURES (",
+                comma_separate(|f| self.doc_display_pass(f), &v.features),
+                ")",
+            ));
+        }
+
+        RcDoc::intersperse(docs, Doc::line()).group()
+    }
+
+    pub(crate) fn doc_create_cluster_replica<'a, T: AstInfo>(
+        &'a self,
+        v: &'a CreateClusterReplicaStatement<T>,
+    ) -> RcDoc<'a> {
+        let mut docs = Vec::new();
+
+        // CREATE CLUSTER REPLICA cluster.replica
+        let replica_name = RcDoc::concat([
+            self.doc_display_pass(&v.of_cluster),
+            RcDoc::text("."),
+            self.doc_display_pass(&v.definition.name),
+        ]);
+        docs.push(nest_title("CREATE CLUSTER REPLICA", replica_name));
+
+        // OPTIONS (...)
+        docs.push(bracket(
+            "(",
+            comma_separate(|o| self.doc_display_pass(o), &v.definition.options),
+            ")",
+        ));
+
+        RcDoc::intersperse(docs, Doc::line()).group()
+    }
+
+    pub(crate) fn doc_create_network_policy<'a, T: AstInfo>(
+        &'a self,
+        v: &'a CreateNetworkPolicyStatement<T>,
+    ) -> RcDoc<'a> {
+        let docs = vec![
+            // CREATE NETWORK POLICY name
+            nest_title("CREATE NETWORK POLICY", self.doc_display_pass(&v.name)),
+            // OPTIONS (...)
+            bracket(
+                "(",
+                comma_separate(|o| self.doc_display_pass(o), &v.options),
+                ")",
+            ),
+        ];
+
+        RcDoc::intersperse(docs, Doc::line()).group()
+    }
+
+    pub(crate) fn doc_create_index<'a, T: AstInfo>(
+        &'a self,
+        v: &'a CreateIndexStatement<T>,
+    ) -> RcDoc<'a> {
+        let mut docs = Vec::new();
+
+        // CREATE [DEFAULT] INDEX [IF NOT EXISTS] [name]
+        let mut title = "CREATE".to_string();
+        if v.key_parts.is_none() {
+            title.push_str(" DEFAULT");
+        }
+        title.push_str(" INDEX");
+        if v.if_not_exists {
+            title.push_str(" IF NOT EXISTS");
+        }
+
+        let index_clause = if let Some(name) = &v.name {
+            self.doc_display_pass(name)
+        } else {
+            RcDoc::nil()
+        };
+        docs.push(nest_title(title, index_clause));
```

Comment:
This comes out a bit awkwardly for indexes with no names:
```
materialize=> select pretty_sql('CREATE DEFAULT INDEX ON foo;');
pretty_sql
CREATE DEFAULT INDEX  ON foo;
```
How about
```
        if let Some(name) = &v.name {
            docs.push(nest_title(title, self.doc_display_pass(name)));
        } else {
            docs.push(RcDoc::text(title));
        }
```
i.e., moving the `docs.push` into the `if` branches?